flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -540,14 +531,13 @@ public void enableCheckpointing(
 
 		checkpointStatsTracker = checkNotNull(statsTracker, "CheckpointStatsTracker");
 
+		CheckpointFailureManager failureManager = new CheckpointFailureManager(chkConfig.getTolerableCheckpointFailureNumber(), () ->
+			failGlobal(new FlinkRuntimeException("Exceeded checkpoint tolerable failure threshold.")));
```

Comment:
Hi @StefanRRichter and @tillrohrmann ping for help, I suspect that this call will cause a deadlock with [this check](https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java#L1196). WDYT? Any opinion?